src/js/background/background.js

```diff
@@ -274,6 +287,88 @@ async function refreshAccountPages() {
   });
 }
 
+// This function is defined as global in the ESLint config _because_ it is created here:
+// eslint-disable-next-line no-redeclare
+async function makeDomainAddress(address, block_list_emails, description = null) {
+  const apiToken = await browser.storage.local.get("apiToken");
+
+  if (!apiToken.apiToken) {
+    browser.tabs.create({
+      url: RELAY_SITE_ORIGIN,
+    });
+    return;
+  }
+
+  const { relayApiSource } = await browser.storage.local.get("relayApiSource");  
+  const serverStoragePermission = await getServerStoragePref();
+  const relayApiUrlRelayAddress = `${relayApiSource}/domainaddresses/`;
+
+  let apiBody = {
+    "enabled": true,
+    "description": "",
+    "block_list_emails": block_list_emails,
+    "used_on": "",
+    "address": address,
+  };
+
+  // Only send description/generated_for/used_on fields in the request if the user is opt'd into server storage
+  if (description && serverStoragePermission) {
+    apiBody.description = description;
+    apiBody.generated_for = description;
+    apiBody.used_on = description + ",";
```

Comment:
Comment added in fe9a02f32e845b816af80dab964dab630b83a7d2

Reply:
Suggestion: Move to a two-dimensional array instead of a brittle CSV layout and transform the data when needed/used. (Will still be stored as a `string` type)

Reply:
Issue filed https://github.com/mozilla/fx-private-relay-add-on/issues/478